### PR TITLE
plugin/loop - ensure zone/qname declared on the right domain

### PR DESCRIPTION
**Pull request by fturib**
### 1. Why is this pull request needed and what does it do?
error (I guess a corner case though) in the setup of plugin loop

### 2. Which issues (if any) are related?
#1995 

### 3. Which documentation changes (if any) need to be made?
None.

### CODE REVIEW

File: plugin/loop/setup.go [hunk 1]
```diff
@@ -73,8 +73,8 @@ func parse(c *caddy.Controller) (*Loop, error) {
 			return nil, c.ArgErr()
 		}
 
-		if len(c.ServerBlockKeys) > 0 {
-			zone = plugin.Host(c.ServerBlockKeys[0]).Normalize()
+		if len(c.Key) > 0 {
```
Comment (miekg) [inline plugin/loop/setup.go:76]: I don't understand this change

Comment (fturib) [reply plugin/loop/setup.go:76]: if you have a server bloc with that header:

domain1, domain2 {
   loop
}

the structure loop will be created twice, one for each Key of the bloc.

the setup of loop will be called 2 times

one with the following informations:

```
c.ServerBlockKeys = ["domain1", "domain2"]
c.Key = "domain1"
```
and second time with:
```
c.ServerBlockKeys = ["domain1", "domain2"]
c.Key = "domain2"
```

if you keep creating the Loop with c.ServerBlockKeys[0], you will always send a query to the domain "domain1" even though you are serving "domain2".

**Comment by miekg** [comment]
[ Quoting <notifications@github.com> in "Re: [coredns/coredns] plugin/loop -..." ]
>if you keep creating the Loop with c.ServerBlockKeys[0], you will always send a query to the domain "domain1" even though you are serving "domain2".

so? This is to detect a loop? Loop detects a loop. If you have advanced config 
that requires multiple queries or what not then I hope you know what you are 
doing.

What does this add? 
Even then I don't understand the change, what's c.Key? No plugin uses this? How 
is this different. Why do we need it here?

**Comment by fturib** [comment]
hum. 
I try to help detecting the loops.

Right now, if you have this config:

```
one.com, two.com {
   loop
   proxy . 8.8.8.8
}
```

If you have a loop on one.com, you will detect it with "loop' plugin.
if you have a loop on two.com,  you will not detect it with "loop" plugin, because it will not send a query to "two.com", but to "one.com".

**Comment by miekg** [comment]
[ Quoting <notifications@github.com> in "Re: [coredns/coredns] plugin/loop -..." ]
>hum.
>I try to help detecting the loops.
>
>Right now, if you have this config:
>
>```
>one.com, two.com {
>   loop
>   proxy . 8.8.8.8
>}
>```
>
>If you have a loop on one.com, you will detect it with "loop' plugin.
>if you have a loop on two.com,  you will not detect it with "loop" plugin, because it will not send a query to "two.com", but to "one.com".

Ok, fair enough. I would say if you don't have a loop for first, you don't have 
a loop period and as Chris has flagged you can still create a loop, that even 
with this change won't be detected. We don't wont to kill the server because we 
send too many queries; so we want to be *very* conservative here.

With the whole looping business: "admin does stupid shit, gets shot in foot", 
you should know better.

**Comment by fturib** [comment]
hum... I understand we do not need this PR.
Closing.

**Pull request closed without merging**